src/pages/cw-multi-test/getting-started/counter/counter-sylvia.mdx

```diff
@@ -0,0 +1,468 @@
+
+**\[dev-dependencies\]**
+
+```toml copy showLineNumbers{23} filename="Cargo.toml"
+[dev-dependencies]
+sylvia = { version = "1", features = ["mt", "cosmwasm_2_0"] }
+```
+
+- **`[dev-dependencies]`** section lists dependencies that are only needed for development and
+  testing.
+- **`sylvia`** the Sylvia framework with features required for testing smart contracts using
+  **`MultiTest`**. Feature `"mt"` in Sylvia should be used only in **[dev-dependencies]** section.
+
+Overall, this `Cargo.toml` file configures a Rust project for a CosmWasm-based smart contract. It
+sets up the basic package details, specifies how the contract should be compiled, defines
+dependencies for core functionality and testing, and includes features to enable or disable certain
+parts of the contract code. This setup ensures the contract can be developed, tested, and deployed
+effectively on the blockchain within the CosmWasm ecosystem.
+
+### `lib.rs`
+
+```rust showLineNumbers filename="lib.rs" copy
+pub mod contract;
+pub mod msg;
+```
+
+The `lib.rs` file in a Rust project serves as the main entry point for defining the structure of a
+library. In the context of our example **counter** smart contract, the `lib.rs` file is defining and
+organizing the modules that make up the contract. Recall the **counter** project file structure:
+
+```ansi {4,6} filename="counter (directory)"
+[34;1m.[0m
+├── Cargo.toml
+└── [34;1msrc[0m
+    ├── contract.rs
+    ├── lib.rs
+    └── msg.rs
+```
+
+There are two modules in the project: `contract.rs` and `msg.rs`. That's why in the `lib.rs` file
+there are two declarations:
+
+- `pub mod contract;{:rust}`<br/> This line declares a public module named **contract**; tells Rust
+  to include the code from a file named `contract.rs` located in the same directory and makes the
+  module publicly accessible (`pub` keyword), which means that other modules or external code can
+  access the entry-points of our smart contract.
+- `pub mod msg;{:rust}`<br/> This line declares a public module named **msg**; includes the code
+  from a file named `msg.rs` and also makes this module public which allows other parts of the code
+  (especially our **counter** smart contract) to access the messages defined here.
+
+Overall, this `lib.rs` file is setting up the main structure of the smart contract by defining its
+key components as separate modules. This organization helps in keeping the code clean, modular, and
+maintainable by separating the core contract logic (`contract.rs`) from the message and query
+definitions (`msg.rs`). This modular approach makes the smart contract easier to understand, extend,
+and test.
+
+### `msg.rs`
+
+The **msg** module in file `msg.rs` typically defines the messages and queries that the smart
+contract accepts and responds to. Messages are usually structured as Rust enums or structs and
+define the input and output interfaces of the contract. In our example this includes messages shown
+below.
```

Comment:
To avoid confusion that there is a need to define your own input messages while using Sylvia.

```suggestion
The **msg** module in file `msg.rs` typically defines responses returned by the contract queries. It is also a convenient place to put any types that might be used in the Contract.
```

Reply:
Ok, but is it also true for `instantiate` entry-point? In this example, can the `CounterInitMsg` be generated automatically?

Reply:
Yes, Sylvia generates here:

```rust
pub struct InstantiateMsg {
    msg: CounterInitMsg,
}
```

It's fine to abstract it to separate type, but if you used `init_value: Option<u32>` instead of the `CounterInitMsg` it would work too.

Reply:
Great but I still have to define `CounterInitMsg` in **msg.rs** file, right?

Reply:
Here is an example of initialization on the blockchain:
``` shell
wasmd tx wasm instantiate 1 '{"msg":{"zero":{}}}' --label my-counter-sv-1 --no-admin --from alice --chain-id wte --keyring-backend=test -o json -y | jq
```
and the schema:
```json
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "InstantiateMsg",
  "type": "object",
  "required": [
    "msg"
  ],
  "properties": {
    "msg": {
      "$ref": "#/definitions/CounterInitMsg"
    }
  },
  "additionalProperties": false,
  "definitions": {
    "CounterInitMsg": {
      "oneOf": [
        {
          "type": "string",
          "enum": [
            "zero"
          ]
        },
        {
          "type": "object",
          "required": [
            "set"
          ],
          "properties": {
            "set": {
              "type": "integer",
              "format": "uint8",
              "minimum": 0.0
            }
          },
          "additionalProperties": false
        }
      ]
    }
  }
}
```

Reply:
And because there is only one `instantiate` entrypoint definition in Sylvia allowed, when someone wants to have multiple variants of instantiating, defining a message is a normal way and is not generated by Sylvia, right?

Reply:
Ok, it's all clear now. The examples will be presented in section with tests for Sylvia contract.